slither/detectors/statements/unused_imports.py

```diff
@@ -0,0 +1,455 @@
+
+    def __init__(
+        self, compilation_unit: SlitherCompilationUnit, slither: "Slither", logger: Logger
+    ):
+        super().__init__(compilation_unit, slither, logger)
+        self.needed_imports = self._new_dict()
+        self.actual_imports = self._new_dict()
+        self.absolute_path_to_imp_filename = {}
+        self.imports_cycle_detected = False
+
+    def _dfs(self, graph: dict, color: dict, x: str):
+        """
+        Simple DFS that checks for a cycle in a given graph.
+        """
+        color[x] = 1
+        for v in graph[x]:
+            if color[v] == 0:
+                self._dfs(graph, color, v)
+            elif color[v] == 1:
+                self.imports_cycle_detected = True
+        color[x] = 2
+
+    def _detect_cycles(self, graph: dict):
+        """
+        Detects cycle in a directed graph.
+        """
+        vertex_color = {}
+        for k in graph.keys():
+            vertex_color[k] = 0
+        for k in graph.keys():
+            for v in graph[k]:
+                if vertex_color[v] == 0:
+                    self._dfs(graph, vertex_color, v)
+                if self.imports_cycle_detected:
+                    return
+
+    def _new_dict(self):
+        """
+        Helper method. Creates a dictionary with all input files as keys and empty sets as values.
+        """
+        dictionary = {}
+        for file in self.compilation_unit.scopes:
+            dictionary[file.absolute] = set()
+        return dictionary
+
+    def _import_to_absolute_path(self, imp: str):
+        """
+        Converts an import to the absolute path of that import.
+        Useful for cases like "import @openzeppelin/...".
+        """
+        for filename in self.compilation_unit.crytic_compile.filenames:
+            if filename.used == imp or filename.relative == imp or filename.short == imp:
+                return filename.absolute
+        return None
+
+    @staticmethod
+    def _add_import(imps: dict, key: str, value: str):
+        """
+        Adds import entries to imps dict.
+        Keys are absolute paths to files (as strings) and each key points to a set of absolute paths (as strings).
+        """
+        if key == value:
+            return
+        imps[key].add(value)
+
+    def _add_custom_type(self, v: Variable):
+        """
+        Adds user defined types to self.needed_imports.
+        """
+        if isinstance(v.type, (TypeAliasTopLevel, TypeAliasContract)):
+            self._add_import(
+                self.needed_imports,
+                v.source_mapping.filename.absolute,
+                v.type.source_mapping.filename.absolute,
+            )
+
+    def _add_item_by_references(self, item: SourceMapping):
+        """
+        Adds all uses of item to self.needed_imports by its references.
+        """
+        for ref in item.references:
+            self._add_import(
+                self.needed_imports, ref.filename.absolute, item.source_mapping.filename.absolute
```

Comment:
You may need to do something like Path(...).absolute() and then use [filename_loop](https://github.com/crytic/crytic-compile/blob/727cd9189fcc3968688461e0e91083972a718327/crytic_compile/crytic_compile.py#L164) in `_import_to_absolute_path`